### PR TITLE
[qfix] Don't start healing if ctx is canceled

### PR DESCRIPTION
**Pull request by Bolodya1997**
## Description
Restore shouldn't start heal if heal context is canceled.

## How Has This Been Tested?

- [ ] Added unit testing to cover
- [x] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [x] Bug fix
- [ ] New functionallity
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

File: pkg/networkservice/common/heal/server.go [hunk 1]
```diff
@@ -193,7 +193,9 @@ func (f *healServer) restoreConnection(ctx context.Context, request *networkserv
 		}
 	}
 
-	f.processHeal(ctx, request.Clone(), opts...)
+	if ctx.Err() == nil {
```
Comment (denis-tingaikin) [inline pkg/networkservice/common/heal/server.go:196]: As I can see we already have a check on `ctx.Err()` at line 217. Should we remove the check from line 217 with this patch?

Comment (Bolodya1997) [reply pkg/networkservice/common/heal/server.go:196]: You mean this for loop condition?
https://github.com/networkservicemesh/sdk/blob/d38c49bc61201044ab9e4060e915171881ba2826/pkg/networkservice/common/heal/server.go#L217

Comment (Bolodya1997) [reply pkg/networkservice/common/heal/server.go:196]: If so, no we shouldn't :)

Comment (Bolodya1997) [reply pkg/networkservice/common/heal/server.go]: Moved this check to `processHeal`.

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**